db/model/subject.js

```diff
@@ -558,30 +513,30 @@ module.exports = function subject(seq, dataTypes) {
           }
         }
 
+        const papChanged = inst.changed('parentAbsolutePath');
+        const pidChanged = inst.changed('parentId');
+        const papEmpty = inst.parentAbsolutePath == null ||
+          inst.parentAbsolutePath == false;
+        const pidEmpty = inst.parentId == null || inst.parentId == false;
```

Comment:
Lines 516 to 520 and Lines 531 to 535 looks the same to me. Do you need both?

Reply:
Yeah, they are both necessary because setupParentFields can change the values and that needs to be reflected in these variables for the logic to be correct.